### PR TITLE
Doc 6429

### PR DESCRIPTION
**Pull request by tonyjhillman**

### CODE REVIEW

**Comment by tonyjhillman** [comment]
I've modified the rebalance.adoc overview, to refer to substages as well as stages. Hopefully what I've done works.

I've removed the description of the rebalance report from add-a-node-and-rebalance, and have pointed to the new reference.

I've added a reference page devoted to explaining the fields of the report and providing examples.

I've updated the REST page, and have broken out the GET for the report, so that it's clear.

**Comment by tonyjhillman** [comment]
This is for DOC-6429 and DOC-6430.

**Comment by tonyjhillman** [comment]
I've also added links to the rebalance reference, on several rest api pages.

File: modules/rebalance-reference/pages/rebalance-reference.adoc [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+The nodes that were intended to be part of the cluster, following rebalance.
+
+** `eject_nodes`.
+The nodes that were to be ejected by rebalance.
+
+** `delta_nodes`.
+The nodes on which Delta Recovery was to be performed.
+
+** `failed_nodes`.
+The nodes for which rebalance failed.
```
Comment (anuthan) [inline modules/rebalance-reference/pages/rebalance-reference.adoc]: These are the nodes that were failed-over prior to rebalance and will be ejected from the cluster on successful rebalance.

File: modules/learn/pages/clusters-and-availability/rebalance.adoc [hunk 2]
```diff
@@ -31,47 +39,65 @@ If and when the missing Data Service node is restored or replaced, rebalance wil
 See xref:learn:clusters-and-availability/intra-cluster-replication.adoc[Intra-Cluster Replication], for information on how data is distributed across nodes.
 
 [#data-service-rebalance-stages]
-=== Data-Service Rebalance Stages
+=== Data-Service Rebalance Substages
 
-vBuckets are moved in _stages_.
-The stages &#8212; which differ, depending on whether the vBucket is an _active_ or a _replica_ vBucket &#8212; are described below.
+During the Data Service rebalance stage, vBuckets are moved in _substages_.
```
Comment (anuthan) [inline modules/learn/pages/clusters-and-availability/rebalance.adoc]: I would prefer the phrasing a vbucket move consists of certain phase(s). The phases depend on whether the vbucket moved is an active or replica vbucket move. 

data-service rebalance has substages example, deltaRecovery substage which signifies the the delta recovery warmup substage of dat service service rebalance.

File: modules/manage/pages/manage-logging/manage-logging.adoc [hunk 3]
```diff
@@ -265,6 +265,11 @@ It is output in common log format and contains several important fields such as
 | `projector`
 | Troubleshooting log for the projector process which is responsible for sending appropriate mutations from Data nodes to Index nodes.
 
+| `rebalance`
+| Contains reports on rebalances that have occurred.
```
Comment (anuthan) [inline modules/manage/pages/manage-logging/manage-logging.adoc:269]: Not all rebalance, currently we store only the past 5 rebalances.

File: modules/rebalance-reference/pages/rebalance-reference.adoc [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+The value is an object, which itself provides a progress figure for each node.
+Each progress figure is specified as a floating-point number between `0.0` and `1.0`.
+In a completed report, the value is expected to be `1`.
+This is an _optional_ field, which may not appear if not implemented for the stage.
+
+* `subStage`.
+Details on any _substages_ that the stage may contain.
+For each _substage_, the standard fields are identical to those for each _stage_.
+This is an _optional_ field, which does not appear, if the stage contains no substages.
+(An example is _delta_recovery_, which is a substage reflected in the `data` output, when Delta Recovery has indeed been performed on one or more nodes running the Data Service.)
```
Comment (anuthan) [inline modules/rebalance-reference/pages/rebalance-reference.adoc]: "when Delta recovery has been performed" -> "when delta recovery Warmup has been performed" .

Unfortunately, delta recovery isn't very clearly understood by everyone hence specifying "warmup" is rather important word that needs to present.

File: modules/rebalance-reference/pages/rebalance-reference.adoc (hunk 1 above)
Comment (tonyjhillman) [reply modules/rebalance-reference/pages/rebalance-reference.adoc]: Done. Thanks.

File: modules/learn/pages/clusters-and-availability/rebalance.adoc (hunk 2 above)
Comment (tonyjhillman) [reply modules/learn/pages/clusters-and-availability/rebalance.adoc]: Okay, I've used 'phase' and 'subphase'. See what you think. I've changed rebalance.adoc and also rebalance-reference.adoc to read in this way.

File: modules/manage/pages/manage-logging/manage-logging.adoc (hunk 3 above)
Comment (tonyjhillman) [reply modules/manage/pages/manage-logging/manage-logging.adoc:269]: Done.

File: modules/rebalance-reference/pages/rebalance-reference.adoc (hunk 4 above)
Comment (tonyjhillman) [reply modules/rebalance-reference/pages/rebalance-reference.adoc]: Done.

**Review by anuthan — APPROVED** [review]

**Review by amarantha-k — APPROVED** [review]

**Pull request merged**